src/anyio/_backends/_asyncio.py

```diff
@@ -763,6 +780,194 @@ def on_completion(task: asyncio.Task[object]) -> None:
+        pass
+
+    def is_closed(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_reader(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_writer(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def remove_signal_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_forever(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_in_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sendfile(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_debug(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def run_until_complete(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_default_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_exception_handler(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def set_task_factory(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def shutdown_asyncgens(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def shutdown_default_executor(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_accept(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_connect(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recv(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recv_into(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recvfrom(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_recvfrom_into(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendall(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendfile(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def sock_sendto(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def start_tls(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def stop(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def subprocess_exec(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def subprocess_shell(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+    def time(self, *args: object, **kwargs: object) -> Any:
+        pass
+
+
+if sys.version_info >= (3, 12):
+    _TF_IS_EAGER_MAP: WeakKeyDictionary[Callable, bool] = WeakKeyDictionary()
+    _TF_IS_EAGER_MAP[asyncio.eager_task_factory] = True
+
+    _TF_CODE = asyncio.eager_task_factory.__code__
+
+    def is_eager(tf: Callable[..., Any] | None) -> bool:
+        if tf is None:
+            return False
+        try:
+            return _TF_IS_EAGER_MAP[tf]
+        except KeyError:
+            pass
+
+        if getattr(tf, "__code__", object()) is _TF_CODE:
+            _TF_IS_EAGER_MAP[tf] = True
+            return True
```

Comment:
Why is it necessary to special case `asyncio.eager_task_factory`?

Reply:
To avoid having to create a task and measure it in the default case